buildinfo_test.go

```diff
@@ -238,19 +248,31 @@ func TestBuildAddDependencies(t *testing.T) {
 }
 
 // Test publish build info without --build-url
-func TestArtifactoryPublishBuildInfo(t *testing.T) {
+func TestArtifactoryPublishAndGetBuildInfo(t *testing.T) {
+	testArtifactoryPublishWithoutBuildUrl(t, tests.RtBuildName1, "10")
+}
+
+// Test publish and get build info with spaces in name.
+func TestArtifactoryPublishAndGetBuildInfoSpecialChars(t *testing.T) {
+	testArtifactoryPublishWithoutBuildUrl(t, tests.RtBuildNameWithSpecialChars, "11")
+}
+
+func testArtifactoryPublishWithoutBuildUrl(t *testing.T, buildName, buildNumber string) {
 	initArtifactoryTest(t)
-	buildNumber := "10"
-	inttestutils.DeleteBuild(artifactoryDetails.Url, tests.RtBuildName1, artHttpDetails)
+	inttestutils.DeleteBuild(artifactoryDetails.Url, buildName, artHttpDetails)
 
-	body := uploadFilesAndGetBuildInfo(t, tests.RtBuildName1, buildNumber, "")
+	bi, err := uploadFilesAndGetBuildInfo(t, buildName, buildNumber, "")
+	if err != nil {
+		return
```

Comment:
Would you like to add an assertNoError here, before the return?
The same goes for a few other places added by this PR.

Reply:
The function asserts the before returning the error. I didn't assert again to avoid duplication. Do you think it is necessary?

Reply:
Since this function is used by only a few similar tests it asserts inside. I added assertions everywhere else